posthog/models/plugin.py

```diff
@@ -340,3 +340,16 @@ def plugin_config_reload_needed(sender, instance, created=None, **kwargs):
 @receiver([post_save, post_delete], sender=PluginAttachment)
 def plugin_attachement_reload_needed(sender, instance, created=None, **kwargs):
     reload_plugins_on_workers()
+
+
+class PluginSourceFile(UUIDModel):
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(name="unique_filename_for_plugin", fields=("plugin_id", "filename")),
+        ]
+
+    plugin: models.ForeignKey = models.ForeignKey("Plugin", on_delete=models.CASCADE)
+    filename: models.CharField = models.CharField(max_length=200, blank=False)
+    source: models.TextField = models.TextField(blank=True, null=True)
```

Comment:
Why can this be null?

Reply:
When the source is actually in a `.zip` file, and we use this model to cache the transpiled output.

Reply:
Let's add a comment to that effect.